src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -178,7 +178,11 @@ def _external_locations(self) -> Iterable[ExternalLocation]:
             curr = queue.pop()
             num_children = len(curr.children)  # 0 - take parent
             if curr.location and (num_children > 1 or num_children == 0):
-                if curr.parent and num_children == 0 and not curr.is_jdbc():  # one table having the prefix
+                # Checking if the parent location is a valid location for external location.
+                # If the table is set to a root location, the root location will be used as the external location.
```

Comment:
Something like this is more accurate?

```suggestion
                # If the current table is a leave node (i.e. no children), its parent's location is used as the external location if that location. 
                # Only works when selecting parent directory for file locations, not for JDBC locations
```

Reply:
KK